### DISCUSSION TITLE
How to install and complier the SNOPT

### DISCUSSION THREAD

**Discussion by xalthzz** [mdolab/pyoptsparse#316] [Installation] [answered]
## Description
I want to install the SNOPT to my pyoptsparse. The [manual](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/optimizers/SNOPT.html#installation) told me as fellow:

> SNOPT is available for purchase [here](http://www.sbsi-sol-optimize.com/asp/sol_snopt.htm). Upon purchase, you should receive a zip file. Within the zip file, there is a folder called src. To use SNOPT with pyoptsparse, paste all files from src except snopth.f into pyoptsparse/pySNOPT/source.

I did it and tested by run `from pyoptsparse.pySNOPT import snopt` in python environment. But it doesn't work. Obviously, there is no "*.so" file in _$/home/mdolabuser/.pyenv/versions/3.8.9/lib/python3.8/site-packages/pyoptsparse/pySNOPT_ .
There is no doubt that I missed some steps(such as complier). But the manual is very short, and I did it step by step. 
So, how can I install and complier SNOPT? Is there some steps not metioned on manual? I think the installation process should not be simple copy and paste.
Thx!

### Steps to reproduce issue

1. Install mdolab [via docker](https://mdolab-mach-aero.readthedocs-hosted.com/en/latest/installInstructions/dockerInstructions.html#pull-mdo-lab-docker-image)
2. Install SNOPT as [manual](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/optimizers/SNOPT.html#installation)  told.
3. Test `from pyoptsparse.pySNOPT import snopt` in python environment.

### Current behavior
It return the error:
> ImportError: cannot import name 'snopt' from 'pyoptsparse.pySNOPT' (/home/mdolabuser/.pyenv/versions/3.8.9/lib/python3.8/site-packages/pyoptsparse/pySNOPT/__init__.py)

### Code versions
- Operating System: Ubuntu 20.04
- Python: 3.8.9
- SNOPT: 7.7.7

**Comment by marcomangano** [comment, marked as answer]
Hello @xalthzz,
have you tried to run `pip install .` from the root `pyOptSparse` folder after you copied the files? We should clarify this in the documentation if it ambiguous.
Let us know if this fixes your issue.

Best,
        Marco
        
P.S. remember also to delete the `meson_build` folder (see tips [here](https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/install.html#debugging-installation-problems)) before installing the package again

**Reply by xalthzz** [reply]
wow！ I forgot to delete `meson_build` berfore.  Now it's working! Thank you!
